source/common/common/win32/thread_impl.cc

```diff
@@ -36,8 +30,8 @@ ThreadPtr ThreadFactoryImplWin32::createThread(std::function<void()> thread_rout
   return std::make_unique<ThreadImplWin32>(thread_routine);
 }
 
-ThreadIdPtr ThreadFactoryImplWin32::currentThreadId() {
-  return std::make_unique<ThreadIdImplWin32>(::GetCurrentThreadId());
+ThreadId ThreadFactoryImplWin32::currentThreadId() {
+  return ThreadId(static_cast<int64_t>(::GetCurrentThreadId()));
```

Comment:
I'm sure this is fine, but @mhoran could you patch and test?

Reply:
I'm going to be out for a week and we haven't yet had a chance to test this out. @wrowe may be able to look at this while I'm away. However, if this needs to be merged in the meantime go ahead, and we'll figure it out after. Thanks!

Reply:
Per conversation at PR level, can you just leave `// TODO(mhoran,wrowe): test this in windows please.` or similar.

Reply:
Done